Backend/lib/shared-services/src/IssueManagement/Storage/CachedQueries/Issue/IssueMessage.hs

```diff
@@ -0,0 +1,109 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# OPTIONS_GHC -Wno-deprecations #-}
+
+module IssueManagement.Storage.CachedQueries.Issue.IssueMessage where
+
+import IssueManagement.Common
+import IssueManagement.Domain.Types.Issue.IssueCategory
+import IssueManagement.Domain.Types.Issue.IssueMessage
+import IssueManagement.Domain.Types.Issue.IssueOption
+import IssueManagement.Domain.Types.Issue.IssueTranslation
+import qualified IssueManagement.Storage.Beam.Issue.IssueMessage as BeamM
+import qualified IssueManagement.Storage.Beam.Issue.IssueTranslation as BeamT
+import IssueManagement.Storage.CachedQueries.CacheConfig
+import qualified IssueManagement.Storage.Queries.Issue.IssueMessage as Queries
+import Kernel.Beam.Lib.UtilsTH
+import Kernel.External.Types (Language)
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import qualified Kernel.Storage.Hedis as Hedis
+import Kernel.Types.Id
+
+findById :: (CacheFlow m r, Esq.EsqDBFlow m r, HasSchemaName BeamM.IssueMessageT) => Id IssueMessage -> Identifier -> m (Maybe IssueMessage)
+findById issueMessageId identifier =
+  Hedis.withCrossAppRedis (Hedis.safeGet $ makeIssueMessageById issueMessageId identifier) >>= \case
+    Just a -> pure a
+    Nothing -> cacheIssueMessageById issueMessageId identifier /=<< Queries.findById issueMessageId
+
+findByIdAndLanguage :: (CacheFlow m r, Esq.EsqDBFlow m r, HasSchemaName BeamM.IssueMessageT, HasSchemaName BeamT.IssueTranslationT) => Id IssueMessage -> Language -> Identifier -> m (Maybe (IssueMessage, Maybe IssueTranslation))
+findByIdAndLanguage issueMessageId language identifier =
+  Hedis.withCrossAppRedis (Hedis.safeGet $ makeIssueMessageByIdAndLanguage issueMessageId language identifier) >>= \case
+    Just a -> pure a
+    Nothing -> cacheIssueMessageByIdAndLanguage issueMessageId language identifier /=<< Queries.findByIdAndLanguage issueMessageId language
+
+findAllByCategoryIdAndLanguage :: (CacheFlow m r, Esq.EsqDBFlow m r, HasSchemaName BeamT.IssueTranslationT, HasSchemaName BeamM.IssueMessageT) => Id IssueCategory -> Language -> Identifier -> m [(IssueMessage, Maybe IssueTranslation)]
```

Comment:
You can create synonym instead of using such long constraints
`Esq.EsqDBFlow m r` is redundant
See my PR for example
https://github.com/nammayatri/nammayatri/pull/3558/files#diff-8bc5123f6b2cfbf117462d9e6c46a89448372edfa493354fe1167684de1c5881R28

Reply:
Done